pkg/install/process.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2021 k0s Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package install
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/mitchellh/go-ps"
+)
+
+func GetProcessID() (pid *int, ppid *int, err error) {
+	processList, err := ps.Processes()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for _, p := range processList {
+		if p.Executable() == "k0s" && hasChildren(p.Pid(), processList) {
+			pid := p.Pid()
+			ppid := p.PPid()
+			return &pid, &ppid, nil
+		}
+	}
+	return nil, nil, nil
+}
+
+func GetProcessOwner(pid int) (string, error) {
+	stdout, err := exec.Command("ps", "-o", "user=", "-p", strconv.Itoa(pid)).Output()
```

Comment:
Future improvement may parse `/proc/[pid]/status` to get the process owner instead of relying of external `ps` utility.

Reply:
Why don't we use `github.com/mitchellh/go-ps` to get PID?

Reply:
> Why don't we use `github.com/mitchellh/go-ps` to get PID?

Does not seem to do what we need. We could use https://pkg.go.dev/github.com/prometheus/procfs#ProcStatus but I think its simpler to just parse it our selves than rely on external dependency in this case.

Reply:
```go
func GetProcessOwner(pid int) (string, error) {
    content, err := os.ReadFile(fmt.Sprintf("/proc/%d/status", pid))
    if err != nil {
        return "", err 
    }
    for _, line := range strings.Split(string(content), "\n") {
        kv := strings.SplitN(line, ":", 2) 
        if kv[0] == "Uid" { 
            return strings.Fields(kv[1])[1], nil
        }
    }
    return "", errors.New("Uid not found in procfs")
}
```

Reply:
Turns out this function is not needed. it is used to find out if "k0s status" is run as root, but this was easily mitigated by using `os.Geteuid()`